pkg/operator/ceph/file/mds/livenessprobe_test.go

```diff
@@ -0,0 +1,339 @@
+package mds
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// simple bash wrapper that provides mock behavior for ceph CLI call(s)
+// acts as shim between go test definitions and bash code under test
+// define as string because embed doesn't work in _test.go files
+var probeShimWrapper string = `
+#!/usr/bin/env bash
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+export SCRIPT_DIR
+function ceph() {
+  # IMPORTANT: tests should always define behavior for specific commands being called to ensure code
+  # changes in the future do not inherit invalid success/failure behavior of existing unit tests
+  if [[ "$1" == "fs" ]] && [[ "$2" == "dump" ]] && [[ "$3" == "--mon-host=" ]] && [[ "$4" == "--mon-initial-members=" ]] && [[ "$5" == "--format" ]] && [[ "$6" == "json" ]]; then
+    [[ -n "$CEPH_MDS_DUMP_MOCK_STDOUT" ]] && echo "$CEPH_MDS_DUMP_MOCK_STDOUT"
+    [[ -n "$CEPH_MDS_DUMP_MOCK_STDERR" ]] && echo "$CEPH_MDS_DUMP_MOCK_STDERR" >/dev/stderr
+    exit $CEPH_MDS_DUMP_MOCK_RETCODE
+  fi
+  echo "command unexpected: ceph $*" >/dev/stderr
+  exit 111 # random return that probably won't overlap with anything else
+}
+export -f ceph
+function jq() {
+	$JQ "$@" # env should specify jq binary location
+}
+export -f jq
+# bash -x $SCRIPT_DIR/livenessprobe.sh
+bash "${@}"
+`
+
+var (
+	//go:embed mdsProbeTestsFiles/livenessprobeTestOutput.json
+	testJsonData string
+	//go:embed mdsProbeTestsFiles/livenessprobeTestOutput2MDS.json
+	testJsonData2MDS string
+	//go:embed mdsProbeTestsFiles/livenessprobeTestOutput0MDS.json
+	testJsonData0MDS string
+)
+
+// writeLines writes the lines to the given file.
+func writeLines(lines []string, path string) (*os.File, error) {
+	file, err := os.Create(path)
+	if err != nil {
+		return file, err
+	}
+	defer file.Close()
+
+	w := bufio.NewWriter(file)
+	for _, line := range lines {
+		fmt.Fprintln(w, line)
+	}
+	return file, w.Flush()
+}
+
+func Test_liveness_probe_script(t *testing.T) {
+	// liveness probe behavior has nuances that are critical to unit test, but there are challenges:
+	// - want to test the script in unit testing so tests live close to implementation
+	// - the probe script is written in bash, so shimming is required to test with go framework
+	// - the script processes output with 'jq'
+	//   -> effective tests will treat 'jq' as a black box to ensure 'jq' commands are correct
+	//   -> it is against good unit testing practice to require CLI tools for tests
+	// - a decent compromise is to skip this test in environments where 'jq' is not present (users)
+	//   BUT ensure the test always runs in CI
+
+	// TODO: set ROOK_UNIT_JQ_PATH in CI
```

Comment:
Unit test are working I am not sure if this is needed

Reply:
```
2023-10-04T14:57:51.8300139Z === RUN   Test_liveness_probe_script
2023-10-04T14:57:51.8300861Z === RUN   Test_liveness_probe_script/0_FS_in_map
2023-10-04T14:57:51.8301771Z === RUN   Test_liveness_probe_script/1_filesystem_and_2_mds_in_map
2023-10-04T14:57:51.8302287Z === RUN   Test_liveness_probe_script/mds_not_in_map
2023-10-04T14:57:51.8302776Z === RUN   Test_liveness_probe_script/filesystem_not_in_map
2023-10-04T14:57:51.8303345Z === RUN   Test_liveness_probe_script/2_filesystem_and_2_MDS_for_each_filesystem_in_map
2023-10-04T14:57:51.8304341Z === RUN   Test_liveness_probe_script/2_filesystem_with_0_MDS_for_first_and_2_MDS_for_second_filesystem_in_map
2023-10-04T14:57:51.8305005Z === RUN   Test_liveness_probe_script/2_filesystem_with_0_MDS_for_first_and_2_MDS_for_second_filesystem_in_map#01
2023-10-04T14:57:51.8305665Z === RUN   Test_liveness_probe_script/2_filesystem_with_1_MDS_for_first_an
```

Reply:
It doesn't matter if it runs today. We need to set this so that we are *certain* the test runs and fails if it stops running. `jq` may be removed from runners in the future, and we need the test to fail if that happens.